x-pack/plugins/ml/public/embeddables/lens/lens_vis_layer_selection_flyout/layer/compatible_layer.tsx

```diff
@@ -0,0 +1,432 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import type * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import React, { FC, useState, useCallback } from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+import useDebounce from 'react-use/lib/useDebounce';
+import type { Embeddable } from '@kbn/lens-plugin/public';
+import type { SharePluginStart } from '@kbn/share-plugin/public';
+import type { DataPublicPluginStart } from '@kbn/data-plugin/public';
+import type { IUiSettingsClient, ApplicationStart } from '@kbn/core/public';
+
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButton,
+  EuiSpacer,
+  EuiIcon,
+  EuiText,
+  EuiFieldText,
+  EuiForm,
+  EuiFormRow,
+  EuiLoadingSpinner,
+  EuiButtonEmpty,
+  EuiCheckbox,
+  EuiAccordion,
+  EuiCallOut,
+} from '@elastic/eui';
+
+import {
+  convertLensToADJob,
+  createAndSaveJob,
+} from '../../../../application/jobs/new_job/job_from_lens';
+import type { LayerResult } from '../../../../application/jobs/new_job/job_from_lens';
+import { CREATED_BY_LABEL, DEFAULT_BUCKET_SPAN } from '../../../../../common/constants/new_job';
+import { extractErrorMessage } from '../../../../../common/util/errors';
+import { MlApiServices } from '../../../../application/services/ml_api_service';
+import { basicJobValidation } from '../../../../../common/util/job_utils';
+import { JOB_ID_MAX_LENGTH } from '../../../../../common/constants/validation';
+import { invalidTimeIntervalMessage } from '../../../../application/jobs/new_job/common/job_validator/util';
+import { ML_APP_LOCATOR } from '../../../../../common/constants/locator';
+
+interface Props {
+  layer: LayerResult;
+  layerIndex: number;
+  embeddable: Embeddable;
+  share: SharePluginStart;
+  data: DataPublicPluginStart;
+  application: ApplicationStart;
+  kibanaConfig: IUiSettingsClient;
+  ml: MlApiServices;
+}
+
+enum STATE {
+  DEFAULT,
+  VALIDATING,
+  SAVING,
+  SAVE_SUCCESS,
+  SAVE_FAILED,
+}
+
+export const CompatibleLayer: FC<Props> = ({
+  layer,
+  layerIndex,
+  embeddable,
+  share,
+  data,
+  application,
+  ml,
+  kibanaConfig,
+}) => {
+  const [jobId, setJobId] = useState('');
+  const [startJob, setStartJob] = useState(true);
+  const [runInRealTime, setRunInRealTime] = useState(true);
+  const [bucketSpan, setBucketSpan] = useState(DEFAULT_BUCKET_SPAN);
+
+  const [jobIdValid, setJobIdValid] = useState<string>('');
+  const [bucketSpanValid, setBucketSpanValid] = useState<string>('');
+  const [state, setState] = useState<STATE>(STATE.DEFAULT);
+  const [createError, setCreateError] = useState<{ text: string; errorText: string } | null>(null);
+
+  function createADJobInWizard() {
+    convertLensToADJob(embeddable, share, layerIndex);
+  }
+
+  async function createADJob() {
+    setState(STATE.SAVING);
+    setCreateError(null);
+    const result = await createAndSaveJob(
```

Comment:
to avoid passing down the services, you can init a `createAndSaveJob` callback (or the class `QuickJobCreator`) with closure as I mentioned above, and put it in a context

Reply:
Updated in https://github.com/elastic/kibana/pull/136421/commits/dc3dc5e003ec704defafc55220c60bc92105b298